experiment/selinux/src/xattrs/xattr.rs

```diff
@@ -0,0 +1,106 @@
+use std::path::Path;
+use rustix::fs as rfs;
+
+#[derive(Debug, thiserror::Error)]
+pub enum XattrError {
+    #[error("Failed to set_xattr: {0}")]
+    SetXattr(String),
+    #[error("Failed to lset_xattr: {0}")]
+    LSetXattr(String),
+    #[error("Failed to get_xattr: {0}")]
+    GetXattr(String),
+    #[error("Failed to lget_xattr: {0}")]
+    LGetXattr(String),    
+}
+
+// function similar with setxattr in golang.org/x/sys/unix repo.
+// set_xattr sets extended attributes on a file specified by its path.
+pub fn set_xattr(fpath: &Path, attr: &str, data: &[u8]) -> Result<(), XattrError> {
+    rfs::setxattr(fpath, attr, data, rfs::XattrFlags::CREATE)
+        .map_err(|e| XattrError::SetXattr(e.to_string()))
+}
+
+// function similar with lsetxattr in golang.org/x/sys/unix repo.
+// lset_xattr sets extended attributes on a symbolic link.
+pub fn lset_xattr(fpath: &Path, attr: &str, data: &[u8]) -> Result<(), XattrError> {
+    rfs::lsetxattr(fpath, attr, data, rfs::XattrFlags::CREATE)
+        .map_err(|e| XattrError::LSetXattr(e.to_string()))
+}
+
+// function similar with getattr in go-selinux repo.
+// get_xattr returns the value of an extended attribute attr set for path.
+pub fn get_xattr(fpath: &Path, attr: &str) -> Result<String, XattrError> {
+    // SELinux label is not so big, so we allocate 1024 bytes for the buffer.
+    let mut buf = vec![0u8; 1024];
+    let size = rfs::getxattr(fpath, attr, &mut buf)
+        .map_err(|e| XattrError::GetXattr(e.to_string()))?;
+    let mut value = String::from_utf8_lossy(&buf[..size]).into_owned();
+    if (!value.is_empty()) && (value.ends_with('\x00')) {
```

Comment:
Shouldn't the \x00 be checked and the buffer size adjusted if it isn't termination.

Reply:
Fixed it. Is this implementation same as your intention?

Reply:
Yes, is this 💯